### ISSUE TITLE
Alternate browser+server API

### ISSUE THREAD

**Issue by wmertens** [kisenka/svg-baker#1]
Would it be possible to use the loader and svg-baker for something like the below?

```js
<SymbolProvider>
    <App/>    // Use with e.g. <UseSymbol symbol={symbol}/>
    <svg>[...all currently-used symbols]</svg> // Gets added automatically, or maybe with another component
</SymbolProvider>
```

`SymbolProvider` would add context so that every `<UseSymbol/>` can register the id as being used.

This way, it also works with server-side rendering, even if multiple render simultaneously. The in-use symbols are all known by the time the `svg` with symbols renders.

**Comment by kisenka** [comment]
Is this something you talking about in this issue https://github.com/kisenka/svg-sprite-loader/issues/45?

**Comment by wmertens** [comment]
Right, I forgot about that one. I took a quick look and it seems to me that I need to make a custom runtime that doesn't do anything, and then just use the objects the loader returns?

**Comment by kisenka** [comment]
Yes, but I don't understand your case entirely, could you describe by code what such runtime should do?

**Comment by wmertens** [comment]
Everything would be done by React components, so there would be no runtime.
Whenever you use a sprite it would pass that to the wrapping SpriteProvider
which would then maintain the svg with the symbol at the end of the body.
